lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -40,7 +40,7 @@ def authorize(amount, payment_method, options = {})
         post[:capture] = false
         build_auth_or_purchase(post, amount, payment_method, options)
 
-        options[:incremental_authorization].to_s.casecmp('true').zero? ? commit(:incremental_authorize, post, payment_method) : commit(:authorize, post)
+        options[:incremental_authorization].to_s.casecmp('true').zero? ? commit(:incremental_authorize, post, options[:payment_id]) : commit(:authorize, post)
```

Comment:
Do we need to use two fields to signify an incremental authorization, or can't we just have a single field called `incremental_authorization`, or maybe `incremental_authorization_id`, and then the presence or absence of a value will indicate whether the customer wishes to perform an incremental authorization:
```
options[:incremental_authorization] ? commit(:incremental_authorize, post, options[:incremental_authorization] : commit(:authorize, post)
```

Reply:
Thanks @dsmcclain , this one looks cleaner. I have updated the change 👍